raiden/message_handler.py

```diff
@@ -163,10 +167,16 @@ def handle_message_refundtransfer(raiden: RaidenService, message: RefundTransfer
                 sender=from_transfer.balance_proof.sender,  # pylint: disable=no-member
             )
             raiden.handle_and_track_state_change(state_change)
+
             if is_light_client:
-                store_refund = StoreRefundTransferLight(message)
-                raiden.handle_and_track_state_change(store_refund)
+                state_change = ActionTransferRerouteLight(
+                    transfer=from_transfer,
+                    secret=random_secret(),
```

Comment:
Ok i understand, so it should be something like this?

`            if is_light_client:
                state_change = ActionTransferRerouteLight(
                    transfer=from_transfer,
                    secret=from_transfer.lock.secrethash,
                    refund_transfer=message
                )
                raiden.handle_and_track_state_change(state_change)`

Reply:
Because if not, i don't know from where i should get the secret hash

Reply:
Not really, you just dont need a secret here. When the lc generates the new payment, it will generate the new secret.

Reply:
Ok and how the node will manage the state change here:

```
def handle_receive_transfer_refund_cancel_route_light(
    chain_state: ChainState, state_change: ActionTransferRerouteLight, storage
) -> TransitionResult[ChainState]:

    new_secrethash = state_change.secrethash
    current_payment_task = chain_state.payment_mapping.secrethashes_to_task[
        state_change.transfer.lock.secrethash
    ]
    chain_state.payment_mapping.secrethashes_to_task.update(
        {new_secrethash: copy.deepcopy(current_payment_task)}
    )
    return subdispatch_to_paymenttask(chain_state, state_change, new_secrethash, storage)
```
I mean i can remove the secret hash but it will never find the payment task unless you tell me that the ```raiden.handle_and_track_state_change(state_change)``` call from above is not triggering this code then i don't get how this is working at all.

Reply:
We should use the refund transfer secret hash to receive the refund transfer into the channel